codex-ui/src/vue/components/alert/useAlert.ts

```diff
@@ -0,0 +1,92 @@
+import type { Ref } from 'vue';
+import { ref } from 'vue';
+import { createSharedComposable } from '@vueuse/core';
+import type { AlertOptions, AlerType } from './Alert.types';
+
+/**
+ *
+ * Return values of useAlert composable
+ */
+export interface UseAlertComposableState {
+  /**
+   * Default alert option
+   */
+  defaultAlertOpt: Ref<AlertOptions>;
+
+  /**
+   * Iterated store of alerts
+   */
+  alerts: Ref<AlertOptions[]>;
+
+  /**
+   * trigger success alert
+   * @param opt - alert options
+   */
+  success: (opt: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger error alert
+   * @param opt - alert options
+   */
+  error: (opt: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger warning alert
+   * @param opt - alert options
+   */
+  warning: (opt: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger info alert
+   * @param opt - alert options
+   */
+  info: (opt: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+
+  /**
+   * trigger default alert
+   * @param opt - alert options
+   */
+  alert: (opt: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>) => void;
+}
+
+/**
+ * Alert service composable hook
+ */
+export const useAlert = createSharedComposable((): UseAlertComposableState => {
+  const alerts = ref<AlertOptions[]>([]);
+
+  const defaultAlertOpt = ref<AlertOptions>({
```

Comment:
redundant duplication. Default props are defined in the Alert.vue

Reply:
fix

Reply:
not fixed